### ISSUE TITLE
Resolving numbers to a contact sometimes changes address dialed

### ISSUE THREAD

**Issue by chadselph** [64characters/Telephone#340]
(Feel free to re-name this ticket to a name that makes more sense to you. I don't know all the terminology.)

Suppose I have a contact in my address book with the number `+1 555 123 4567`.
Then, suppose I want to use a specific SIP trunking account to place a call, if I type: `sip:+15551234567@mysiptrunk.com`. When I initiate the call, the app actually resolves this to the above contact and changes my call to `Contact Name <+15551234567>` and doesn't use `mysiptrunk.com`.

I know this is probably an edge case, but it would be nice for this functionality to be possible. My current work-around is to remove the `+1` from my contact and then add it back afterwards. I'm on version 1.2.6.

**Comment by eofster** [comment]
Right, the host part is being stripped off during the contact matching phase. I’ll need to dive deeper into the question if it was done on purpose or by accident. In the meantime, could you tell me if configuring a separate account in Telephone for another trunk with the corresponding domain name that is used by default in such cases would be a solution for you? As I see it, In this case you won’t need to manually type in the trunk’s domain name, but rather just pick the right account window in the app.

**Comment by chadselph** [comment]
Yeah, it might be possible, I need to learn some SIP terminology and then play with the settings.

The server I'm using for trunking doesn't have registration, so the UI won't go "Available". I tried creating an account with Service-1 as the "Registry Server" and Service-2 as the "Domain" but this didn't work either.

I realize it's not a normal usage pattern, I'm mostly just playing around to learn about SIP.

**Comment by chadselph** [comment]
Also, I can't quite figure out exactly when it does matching and when it doesn't. It seems like a call sometimes triggers the match, but other times it just uses the full address.

**Comment by eofster** [comment]
If each of your servers doesn’t make any difference between the domain part of the SIP address it expects and the server’s hostname that should be used as a Registrar, than you don’t have to fill out the Registrar field in Telephone. The Registrar’s hostname will be taken from the Domain field.

What I had in mind is that you configure two accounts in Telephone: one with Service-1 domain, another one with Service-2 domain and leave Registrar fields empty if they match their corresponding domains. Then, when calling, you never specify the host past in Telephone’s call destination field, but rather pick the appropriate account window.

The matching is normally triggered automatically by the text field on certain events. This includes explicit events like pressing the Return button, but also the events when the window becomes active. You might have seen it working (not stripping out the host part) when SIP address you were typing in didn’t match any address book records. In this case it should work as you originally expected it to.

**Comment by chadselph** [comment]
Service-2 doesn't have SIP registration, so while I can send INVITEs to it, I can't get the Telephone UI to see it as online/Available
